syntax = "proto3";

package project;

service DataExchange {
    rpc SendData (DataChunk) returns (Response);
    rpc ReceiveData (Empty) returns (stream DataChunk);
    rpc GetData (Request) returns (DataChunk);
    rpc ListData (Empty) returns (DataList);
}

message DataChunk {
    bytes data = 1;
    string file_name = 2;
}

message Response {
    string message = 1;
}

message Request {
    string file_name = 1;
}

message DataList {
    repeated string file_names = 1;
}

message Empty {}
